pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -515,26 +511,26 @@ class ModelValidatorImpl implements ModelValidator {
             && !opt.args.every { it instanceof ModelASTKeyValueOrMethodCallPair }) {
             errorCollector.error(opt, Messages.ModelValidatorImpl_MixedNamedAndUnnamedParameters())
             valid = false
-        } else {
-            valid = validateElement((ModelASTMethodCall)opt)
         }
         return valid
     }
 
     private boolean validateParameterType(ModelASTValue v, Class erasedType, ModelASTKey k = null) {
         if (v.isLiteral()) {
             try {
-                // Converting from boolean or int to string at runtime doesn't work, but does pass castToType. So.
-                if (erasedType.equals(String.class)
-                    && (v.value instanceof Integer || v.value instanceof Boolean)) {
+                // Converting amongst boolean, string and int at runtime doesn't work, but does pass castToType. So.
+                if ((erasedType.equals(String.class) && (v.value instanceof Integer || v.value instanceof Boolean)) ||
+                    (erasedType.equals(int.class) && (v.value instanceof String || v.value instanceof Boolean))) {
```

Comment:
e.g. `(erasedType.equals(Integer.class)` would be the same but without the autoboxing?